.github/workflows/check_docs.yml

```
@@ -0,0 +1,49 @@
+name: Check Docs
+
+on:
+  pull_request_target:
+    branches:
+      - main
+  pull_request:
+
+env:
+  RAILS_ENV: test
+
+jobs:
+  check-docs:
+    needs: labeler
+    if: contains(github.event.pull_request.labels.*.name, 'doc')
```

I think this `if` is preventing it to run on `push` in the main branch

```suggestion
    if: !github.event.pull_request || contains(github.event.pull_request.labels.*.name, 'doc')
```